src/main/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepository.java

```diff
@@ -172,31 +181,39 @@ public CaseTypeDefinitionVersion doGetLatestVersion(String caseTypeId) {
     }
 
     @Override
-    @Cacheable("caseTypeDefinitionsCache")
-    public CaseType getCaseType(int version, String caseTypeId) {
-        return this.getCaseType(caseTypeId);
+    public List<Jurisdiction> getJurisdictions(List<String> ids) {
+        ArrayList<Jurisdiction> jurisdictions = new ArrayList<>(ids.size());
```

Comment:
this method can potentially be called bypassing caching IF this repository is injected without wrapping it with the cached repository. In that case this implementation has got performance issues. Because it makes many single calls rather than one. I would suggest we reimplement this to make 1 http call only that fetches all the jurisdictions. And in the cached repository we keep the current implementation that loops and call the getJurisdiction and triggers that caching

Reply:
Done as discussed. Thank you.